src/libraries/System.DirectoryServices.AccountManagement/src/System/DirectoryServices/AccountManagement/Context.cs

```diff
@@ -44,7 +44,7 @@ internal static class CapabilityMap
         public const string LDAP_CAP_ACTIVE_DIRECTORY_V61_OID = "1.2.840.113556.1.4.1935";
     }
 
-    internal sealed class CredentialValidator
+    internal sealed class CredentialValidator : IDisposable
```

Comment:
Even though you are making the class disposable here, you arent' actually using it on the places where it gets instantiated (In PrincipalContext constructor). In order for this to actually get disposed you need to add a `_credValidate.Dispose()` call on PrincipalContext.Dispose()

Reply:
Thanks for reviewing my PR! I did add a [`_credValidate.Dispose()` call](https://github.com/dotnet/runtime/blob/810ac4b8275b2da32a084c770474263df3697f49/src/libraries/System.DirectoryServices.AccountManagement/src/System/DirectoryServices/AccountManagement/Context.cs#L1025) on PrincipalContext.Dispose() in this PR (line 1025 in Context.cs). Since the only place where `CredentialValidator` is instantiated is in the constructor of PrincipalContext, this should ensure that it gets properly disposed. Or am I missing something here?

Reply:
You are totally right, the one that missed it was me, sorry 😄. I was thrown off by the fact that for these types we don't have separate files.